### ISSUE TITLE
[JENKINS-7641] Slaves hang when archiving artifacts

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/ssh-agents-plugin#689]
<p>I don't know why this happens, but my slaves have begun to hang when they get to the Archiving Artifacts portion of my job:</p>

```
Archiving artifacts
ERROR: Failed to archive artifacts: dist/**
hudson.util.IOException2: Failed to extract /mnt/hudsonslave/workspace/simplegeo-puppet-manifests/dist/**
	at hudson.FilePath.readFromTar(FilePath.java:1577)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:1491)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:117)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:19)
	at hudson.model.AbstractBuild$AbstractRunner.perform(AbstractBuild.java:601)
	at hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:580)
	at hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:558)
	at hudson.model.Build$RunnerImpl.post2(Build.java:157)
	at hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:528)
	at hudson.model.Run.run(Run.java:1303)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:137)
Caused by: java.io.IOException
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:173)
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
	at java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:221)
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:141)
	at java.util.zip.GZIPInputStream.read(GZIPInputStream.java:92)
	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)
	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)
	at hudson.org.apache.tools.tar.TarInputStream.read(TarInputStream.java:345)
	at java.io.FilterInputStream.read(FilterInputStream.java:90)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1025)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:999)
	at hudson.util.IOUtils.copy(IOUtils.java:33)
	at hudson.FilePath.readFromTar(FilePath.java:1565)
	... 12 more

```

<p>I aborted this job, so I don't know if the error is related or not. The job runs fine, but it hangs during archiving. If I restart the connection to the node (in /computer/; not restarting Hudson or the node itself), jobs will build successfully again for a while.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ieure.jpg" title="ieure's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ieure">ieure</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641" target="_blank">Slaves hang when archiving artifacts</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ifernandezcalvo.png" title="ifernandezcalvo's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ifernandezcalvo">ifernandezcalvo</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: ssh-slaves-plugin
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2018-12-28T16:42:03+00:00
<li><b>votes</b>: 49
<li><b>watchers</b>: 57
<li><b>imported</b>: 20251211-223321
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I don't know why this happens, but my slaves have begun to hang when they get to the Archiving Artifacts portion of my job:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Archiving artifacts
ERROR: Failed to archive artifacts: dist/**
hudson.util.IOException2: Failed to extract /mnt/hudsonslave/workspace/simplegeo-puppet-manifests/dist/**
	at hudson.FilePath.readFromTar(FilePath.java:1577)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:1491)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:117)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:19)
	at hudson.model.AbstractBuild$AbstractRunner.perform(AbstractBuild.java:601)
	at hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:580)
	at hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:558)
	at hudson.model.Build$RunnerImpl.post2(Build.java:157)
	at hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:528)
	at hudson.model.Run.run(Run.java:1303)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:137)
Caused by: java.io.IOException
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:173)
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
	at java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:221)
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:141)
	at java.util.zip.GZIPInputStream.read(GZIPInputStream.java:92)
	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)
	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)
	at hudson.org.apache.tools.tar.TarInputStream.read(TarInputStream.java:345)
	at java.io.FilterInputStream.read(FilterInputStream.java:90)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1025)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:999)
	at hudson.util.IOUtils.copy(IOUtils.java:33)
	at hudson.FilePath.readFromTar(FilePath.java:1565)
	... 12 more
</pre>
</div></div>

<p>I aborted this job, so I don't know if the error is related or not. The job runs fine, but it hangs during archiving. If I restart the connection to the node (in /computer/; not restarting Hudson or the node itself), jobs will build successfully again for a while.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Ubuntu Server 10.04 64-bit</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nnau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=169514&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-169514">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>also see this error - it's impossible to archive artifacts on ia64 slave (redhat linux) - just hangs. 
files a separate issue for that: <a href="https://issues.jenkins-ci.org/browse/JENKINS-15682" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-15682</a></p>
</pre>
</details>

</ul>
</details>

<p>also see this error - it's impossible to archive artifacts on ia64 slave (redhat linux) - just hangs. <br/>
files a separate issue for that: <a href="https://issues.jenkins-ci.org/browse/JENKINS-15682" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-15682</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>brevilo:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=174575&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174575">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote>
<p>switching to Oracle's JDK from Open JDK</p></blockquote>
<p>Unfortunately that's not an option for OSX &lt; 10.7</p>
</pre>
</details>

</ul>
</details>

<blockquote>
<p>switching to Oracle's JDK from Open JDK</p></blockquote>
<p>Unfortunately that's not an option for OSX < 10.7</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>aswini_r:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=179681&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179681">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi, I use jenkins version 1.514 and this issues happens for me for the past 1 week. This is a big blocker for me in executing builds in slave.
I have installed jenkins on linux box and all the slaves are also remote linux boxes connected via SSH.
Can anyone advise how to resolve this at the earliest? Thanks a mil for your help in advance.</p>

<p>Regards,
Aswini</p>
</pre>
</details>

</ul>
</details>

<p>Hi, I use jenkins version 1.514 and this issues happens for me for the past 1 week. This is a big blocker for me in executing builds in slave.<br/>
I have installed jenkins on linux box and all the slaves are also remote linux boxes connected via SSH.<br/>
Can anyone advise how to resolve this at the earliest? Thanks a mil for your help in advance.</p>

<p>Regards,<br/>
Aswini</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pfalcon:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=195022&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-195022">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Ok, so this is probably as old as Jenkins itself, and probably ready to become (if not already) its businesscard - what's that Jenkins thing? Ah, it does some CI and bunch of server hanging around. And the biggest problem is not that the issue described here happens, it's what shown by this log:</p>

<p>01:53:56.709 Archiving artifacts
12:02:21.094 ERROR: Failed to archive artifacts: *.sum
12:02:21.096 hudson.util.IOException2: hudson.util.IOException2: Failed to extract /home/buildslave/workspace/cbuild/transfer of 1 files
12:02:21.096 Caused by: java.io.IOException
12:02:21.096 	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:175)</p>

<p>So, it hanged build server for 10hrs. As it usually happens, there're more jobs accumulate for this build slave, other builds waiting for this build to complete, so entire CI system comes to a halt, followed by kaboom. And if nobody's looking, Jenkins will happily and shamelessly lock up your servers for days.</p>

<p>So, why this happens? Surely, there's no single reason, but carefully selected assorti of toxic stuff:</p>

<p>1. Bugs in JVM/Java - just because there're comments above that switching to another JDK version/provider seem to alleviate it.
2. Bugs in Jenkins - just because every new release brings only security fixes as if previous version was something like 0.0.1, so you may imagine there's lot to fix yet.
3. But most importantly, that's the way Jenkins is written. To illustrate that, let's look at linked <a href="https://issues.jenkins.io/browse/JENKINS-11586" title="Regression introduced with Slave Side ChannelPinger" class="issue-link" data-issue-key="JENKINS-11586"><del>JENKINS-11586</del></a>, comment straight from Kohsuke: "The ping currently is supposed to wait for 4 minutes before it gives up and kills the channel. But I have a hard time believing that the channel did really clog for 4 minutes." Here we go. Network over which that channel goes is UNRELIABLE. Everything you find hard to believe about it is actually true. It may fail to deliver stuff, it may clog for 4 or 40 minutes, RIAA may knock on your door telling you download forbidden torrents which you never did. Or take another example, straight from FastPipedInputStream.java as quoted in stacktrace above. In its header, it confesses to be java.io.PipedInputStream equivalent, which just "uses proper synchronization" and "doesn't rely on polling". A seasonal engineer would recognize smartness and forthlooking of Java engineers who did not trust Java to do synchronization and instead used strings-and-stick method. Now smart kids came who thought they could make it better, and now their code locks up servers throughout the world.</p>

<p>Ok, enough intro to problem area. Let's look straight into FastPipedInputStream.java:175 which was caught red-handed in the stacktrace above: <a href="https://github.com/jenkinsci/remoting/blob/master/src/main/java/hudson/remoting/FastPipedInputStream.java#L163" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/remoting/blob/master/src/main/java/hudson/remoting/FastPipedInputStream.java#L163</a>
And what we see immediately is infinite loop. Just see yourself: it blocks in wait on buffer for 10s, then does some liveness checks (which, as we already learned, will fail to detect any issues regularly), the checks for updates from outside, and if nothing happens, it will hang there forever. Here it is, Jenkins coding style.</p>

<p>Here's the patch: <a href="https://github.com/pfalcon/remoting/commit/239b3dcf26498ff296fabf770dffc7a456b2878c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/pfalcon/remoting/commit/239b3dcf26498ff296fabf770dffc7a456b2878c</a></p>

<p>So, why am I writing all this? Over time, I learned that if people come to me with artifact archiving issues, the best thing I can suggest them is AVOID NATIVE JENKINS ARTIFACT ARCHIVING LIKE A PLAGUE. People listen, and that job whose stack trace is quoted above is no longer uses it (rest of our jobs didn't use it for years). So, while I hacked up the patch above, I don't really have sandbox to test it with. So, if you experience this issue, I encourage you to try that patch and share results. Just to clarify - the patch above is not going to fix this issue (if you want it not fail, then Java and Jenkins are wrong technologies). But instead it makes it fail fast and not waste the resources (it also addresses only one infinite loop, I'm sure there're dozens more).</p>

<p>The stacktrace above happened with Jenins 1.532.1 on Linux/Ubuntu master/slave (x86, x64, arm slaves affected).</p>

<p>Thanks for listening to the rant, and happy (or sour) Jenkinsing!</p>
</pre>
</details>

</ul>
</details>

<p>Ok, so this is probably as old as Jenkins itself, and probably ready to become (if not already) its businesscard - what's that Jenkins thing? Ah, it does some CI and bunch of server hanging around. And the biggest problem is not that the issue described here happens, it's what shown by this log:</p>

<p>01:53:56.709 Archiving artifacts<br/>
12:02:21.094 ERROR: Failed to archive artifacts: *.sum<br/>
12:02:21.096 hudson.util.IOException2: hudson.util.IOException2: Failed to extract /home/buildslave/workspace/cbuild/transfer of 1 files<br/>
12:02:21.096 Caused by: java.io.IOException<br/>
12:02:21.096 	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:175)</p>

<p>So, it hanged build server for 10hrs. As it usually happens, there're more jobs accumulate for this build slave, other builds waiting for this build to complete, so entire CI system comes to a halt, followed by kaboom. And if nobody's looking, Jenkins will happily and shamelessly lock up your servers for days.</p>

<p>So, why this happens? Surely, there's no single reason, but carefully selected assorti of toxic stuff:</p>

<p>1. Bugs in JVM/Java - just because there're comments above that switching to another JDK version/provider seem to alleviate it.<br/>
2. Bugs in Jenkins - just because every new release brings only security fixes as if previous version was something like 0.0.1, so you may imagine there's lot to fix yet.<br/>
3. But most importantly, that's the way Jenkins is written. To illustrate that, let's look at linked <a href="https://issue-redirect.jenkins.io/issue/11586" title="Regression introduced with Slave Side ChannelPinger" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/11586">JENKINS-11586</a>"><del>JENKINS-11586</del></a>, comment straight from Kohsuke: "The ping currently is supposed to wait for 4 minutes before it gives up and kills the channel. But I have a hard time believing that the channel did really clog for 4 minutes." Here we go. Network over which that channel goes is UNRELIABLE. Everything you find hard to believe about it is actually true. It may fail to deliver stuff, it may clog for 4 or 40 minutes, RIAA may knock on your door telling you download forbidden torrents which you never did. Or take another example, straight from FastPipedInputStream.java as quoted in stacktrace above. In its header, it confesses to be java.io.PipedInputStream equivalent, which just "uses proper synchronization" and "doesn't rely on polling". A seasonal engineer would recognize smartness and forthlooking of Java engineers who did not trust Java to do synchronization and instead used strings-and-stick method. Now smart kids came who thought they could make it better, and now their code locks up servers throughout the world.</p>

<p>Ok, enough intro to problem area. Let's look straight into FastPipedInputStream.java:175 which was caught red-handed in the stacktrace above: <a href="https://github.com/jenkinsci/remoting/blob/master/src/main/java/hudson/remoting/FastPipedInputStream.java#L163" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/remoting/blob/master/src/main/java/hudson/remoting/FastPipedInputStream.java#L163</a><br/>
And what we see immediately is infinite loop. Just see yourself: it blocks in wait on buffer for 10s, then does some liveness checks (which, as we already learned, will fail to detect any issues regularly), the checks for updates from outside, and if nothing happens, it will hang there forever. Here it is, Jenkins coding style.</p>

<p>Here's the patch: <a href="https://github.com/pfalcon/remoting/commit/239b3dcf26498ff296fabf770dffc7a456b2878c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/pfalcon/remoting/commit/239b3dcf26498ff296fabf770dffc7a456b2878c</a></p>

<p>So, why am I writing all this? Over time, I learned that if people come to me with artifact archiving issues, the best thing I can suggest them is AVOID NATIVE JENKINS ARTIFACT ARCHIVING LIKE A PLAGUE. People listen, and that job whose stack trace is quoted above is no longer uses it (rest of our jobs didn't use it for years). So, while I hacked up the patch above, I don't really have sandbox to test it with. So, if you experience this issue, I encourage you to try that patch and share results. Just to clarify - the patch above is not going to fix this issue (if you want it not fail, then Java and Jenkins are wrong technologies). But instead it makes it fail fast and not waste the resources (it also addresses only one infinite loop, I'm sure there're dozens more).</p>

<p>The stacktrace above happened with Jenins 1.532.1 on Linux/Ubuntu master/slave (x86, x64, arm slaves affected).</p>

<p>Thanks for listening to the rant, and happy (or sour) Jenkinsing!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pfalcon:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=196148&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-196148">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Woo-hoo, the patch in the previous comment works:</p>

<p>00:18:59.873 FATAL: HTML Publisher failure
00:18:59.873 hudson.util.IOException2: hudson.util.IOException2: Failed to extract /srv/jenkins/workspace/job/doc/html/*<b>/</b>
00:18:59.873 	at hudson.FilePath.readFromTar(FilePath.java:2066)
00:18:59.873 	at hudson.FilePath.copyRecursiveTo(FilePath.java:1978)
00:18:59.873 	at hudson.FilePath.copyRecursiveTo(FilePath.java:1889)
00:18:59.873 	at hudson.FilePath.copyRecursiveTo(FilePath.java:1872)
00:18:59.873 	at htmlpublisher.HtmlPublisher.perform(HtmlPublisher.java:213)
00:18:59.873 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
00:18:59.873 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:785)
00:18:59.873 	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:757)
00:18:59.873 	at hudson.model.Build$BuildExecution.post2(Build.java:183)
00:18:59.873 	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:706)
00:18:59.873 	at hudson.model.Run.execute(Run.java:1690)
00:18:59.873 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
00:18:59.873 	at hudson.model.ResourceController.execute(ResourceController.java:88)
00:18:59.873 	at hudson.model.Executor.run(Executor.java:246)
00:18:59.873 Caused by: java.io.IOException: High-level timeout waiting for activity on pipe
00:18:59.873 	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:180)
00:18:59.873 	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
00:18:59.873 	at com.jcraft.jzlib.InflaterInputStream.fill(InflaterInputStream.java:175)
00:18:59.873 	at com.jcraft.jzlib.InflaterInputStream.read(InflaterInputStream.java:106)
00:18:59.873 	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)
00:18:59.873 	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)
00:18:59.873 	at hudson.org.apache.tools.tar.TarInputStream.getNextEntry(TarInputStream.java:228)
00:18:59.873 	at hudson.FilePath.readFromTar(FilePath.java:2044)
00:18:59.873 	... 13 more</p>


<p>So, the timeout as it is now is 1000s. Trying to cut that in 2 and see if any false positive timeouts would happen.</p>

</pre>
</details>

</ul>
</details>

<p>Woo-hoo, the patch in the previous comment works:</p>

<p>00:18:59.873 FATAL: HTML Publisher failure<br/>
00:18:59.873 hudson.util.IOException2: hudson.util.IOException2: Failed to extract /srv/jenkins/workspace/job/doc/html/*<b>/</b><br/>
00:18:59.873 	at hudson.FilePath.readFromTar(FilePath.java:2066)<br/>
00:18:59.873 	at hudson.FilePath.copyRecursiveTo(FilePath.java:1978)<br/>
00:18:59.873 	at hudson.FilePath.copyRecursiveTo(FilePath.java:1889)<br/>
00:18:59.873 	at hudson.FilePath.copyRecursiveTo(FilePath.java:1872)<br/>
00:18:59.873 	at htmlpublisher.HtmlPublisher.perform(HtmlPublisher.java:213)<br/>
00:18:59.873 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)<br/>
00:18:59.873 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:785)<br/>
00:18:59.873 	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:757)<br/>
00:18:59.873 	at hudson.model.Build$BuildExecution.post2(Build.java:183)<br/>
00:18:59.873 	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:706)<br/>
00:18:59.873 	at hudson.model.Run.execute(Run.java:1690)<br/>
00:18:59.873 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
00:18:59.873 	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
00:18:59.873 	at hudson.model.Executor.run(Executor.java:246)<br/>
00:18:59.873 Caused by: java.io.IOException: High-level timeout waiting for activity on pipe<br/>
00:18:59.873 	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:180)<br/>
00:18:59.873 	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)<br/>
00:18:59.873 	at com.jcraft.jzlib.InflaterInputStream.fill(InflaterInputStream.java:175)<br/>
00:18:59.873 	at com.jcraft.jzlib.InflaterInputStream.read(InflaterInputStream.java:106)<br/>
00:18:59.873 	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)<br/>
00:18:59.873 	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)<br/>
00:18:59.873 	at hudson.org.apache.tools.tar.TarInputStream.getNextEntry(TarInputStream.java:228)<br/>
00:18:59.873 	at hudson.FilePath.readFromTar(FilePath.java:2044)<br/>
00:18:59.873 	... 13 more</p>


<p>So, the timeout as it is now is 1000s. Trying to cut that in 2 and see if any false positive timeouts would happen.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>generatz:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=198160&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-198160">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Like mjmac, hangs after "Archiving artifacts" if I don't have "Response Time" checked under "Preventive Node Monitoring" for my slave nodes. However, if the monitoring is enabled, I get the IOException. Happens more often than not, but is intermittent. If I use my Jenkins master (jenkins-1.549-1.1; Intel server with RHEL 6.4) as a slave I don't see any issues, ever. My other slaves are IBM PPC servers (also with RHEL6.4) - that's where I see the problem. I've only tested with a 20-byte file, so size isn't the issue.</p>

<p>If I kill a hung job I get the following:
Archiving artifacts
ERROR: Failed to archive artifacts: the_date.txt
java.io.IOException: java.io.IOException: Failed to extract /bgusr/home1/jenkins/bgqmsn0/workspace/gene_triggerer2/transfer of 1 files
	at hudson.FilePath.readFromTar(FilePath.java:2088)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2000)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:57)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:140)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:784)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:756)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:705)
	at hudson.model.Run.execute(Run.java:1695)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:231)
Caused by: java.io.IOException
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:175)
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
	at com.jcraft.jzlib.InflaterInputStream.fill(InflaterInputStream.java:175)
	at com.jcraft.jzlib.InflaterInputStream.read(InflaterInputStream.java:106)
	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)
	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)
	at hudson.org.apache.tools.tar.TarInputStream.getNextEntry(TarInputStream.java:228)
	at hudson.FilePath.readFromTar(FilePath.java:2066)
	... 12 more</p>

<p>	at hudson.FilePath.copyRecursiveTo(FilePath.java:2007)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:57)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:140)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:784)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:756)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:705)
	at hudson.model.Run.execute(Run.java:1695)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:231)
Caused by: java.util.concurrent.ExecutionException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request$1.get(Request.java:278)
	at hudson.remoting.Request$1.get(Request.java:210)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:59)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2003)
	... 11 more
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.abort(Request.java:299)
	at hudson.remoting.Channel.terminate(Channel.java:782)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2570)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1314)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:368)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)</p>
</pre>
</details>

</ul>
</details>

<p>Like mjmac, hangs after "Archiving artifacts" if I don't have "Response Time" checked under "Preventive Node Monitoring" for my slave nodes. However, if the monitoring is enabled, I get the IOException. Happens more often than not, but is intermittent. If I use my Jenkins master (jenkins-1.549-1.1; Intel server with RHEL 6.4) as a slave I don't see any issues, ever. My other slaves are IBM PPC servers (also with RHEL6.4) - that's where I see the problem. I've only tested with a 20-byte file, so size isn't the issue.</p>

<p>If I kill a hung job I get the following:<br/>
Archiving artifacts<br/>
ERROR: Failed to archive artifacts: the_date.txt<br/>
java.io.IOException: java.io.IOException: Failed to extract /bgusr/home1/jenkins/bgqmsn0/workspace/gene_triggerer2/transfer of 1 files<br/>
	at hudson.FilePath.readFromTar(FilePath.java:2088)<br/>
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2000)<br/>
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:57)<br/>
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:140)<br/>
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:784)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:756)<br/>
	at hudson.model.Build$BuildExecution.post2(Build.java:183)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:705)<br/>
	at hudson.model.Run.execute(Run.java:1695)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
	at hudson.model.Executor.run(Executor.java:231)<br/>
Caused by: java.io.IOException<br/>
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:175)<br/>
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)<br/>
	at com.jcraft.jzlib.InflaterInputStream.fill(InflaterInputStream.java:175)<br/>
	at com.jcraft.jzlib.InflaterInputStream.read(InflaterInputStream.java:106)<br/>
	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)<br/>
	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)<br/>
	at hudson.org.apache.tools.tar.TarInputStream.getNextEntry(TarInputStream.java:228)<br/>
	at hudson.FilePath.readFromTar(FilePath.java:2066)<br/>
	... 12 more</p>

<p>	at hudson.FilePath.copyRecursiveTo(FilePath.java:2007)<br/>
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:57)<br/>
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:140)<br/>
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:784)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:756)<br/>
	at hudson.model.Build$BuildExecution.post2(Build.java:183)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:705)<br/>
	at hudson.model.Run.execute(Run.java:1695)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
	at hudson.model.Executor.run(Executor.java:231)<br/>
Caused by: java.util.concurrent.ExecutionException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.Request$1.get(Request.java:278)<br/>
	at hudson.remoting.Request$1.get(Request.java:210)<br/>
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:59)<br/>
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2003)<br/>
	... 11 more<br/>
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.Request.abort(Request.java:299)<br/>
	at hudson.remoting.Channel.terminate(Channel.java:782)<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)<br/>
Caused by: java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)<br/>
Caused by: java.io.EOFException<br/>
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2570)<br/>
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1314)<br/>
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:368)<br/>
	at hudson.remoting.Command.readFrom(Command.java:92)<br/>
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>valkolovos:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=199920&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199920">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This only became a problem for me when I upgraded from 556 to 561, but it now hangs on every slave build.</p>
</pre>
</details>

</ul>
</details>

<p>This only became a problem for me when I upgraded from 556 to 561, but it now hangs on every slave build.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=199921&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199921">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=valkolovos" class="user-hover" rel="valkolovos">valkolovos</a>: You're experiencing the unrelated and newly introduced in 1.560 <a href="https://issues.jenkins.io/browse/JENKINS-22734" title="OutOfMemory on file transfer in 1.560" class="issue-link" data-issue-key="JENKINS-22734"><del>JENKINS-22734</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=valkolovos" class="user-hover" rel="valkolovos">valkolovos</a>: You're experiencing the unrelated and newly introduced in 1.560 <a href="https://issue-redirect.jenkins.io/issue/22734" title="OutOfMemory on file transfer in 1.560" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/22734">JENKINS-22734</a>"><del>JENKINS-22734</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/inb.png" title="inb's avatar" /> inb:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=204128&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204128">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm getting this about once every three or four builds now, mostly with windows (cygwin) slaves but not exclusively. My artifacts range from 1-2 mb to 800mb. Size appears not to be an issue.</p>

<p>At the time things hang (forever) both master and slave become quiet. Other jobs on other slaves progress without issue.</p>

<p>When I cancel the stuck build I see:</p>

<p>ERROR: Failed to archive artifacts: pool.tar.gz,*<b>/</b>.log,built.<b>,pool/</b><b>/</b>-Results.xml,<b>_pool.tar.gz,</b>.html,*.st
java.io.IOException: java.io.IOException: Failed to extract .....
&lt;snip&gt;
Caused by: java.io.IOException
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:177)
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
&lt;snip&gt;</p>

<p>Will pfalcon's patch cure this?</p>
</pre>
</details>

</ul>
</details>

<p>I'm getting this about once every three or four builds now, mostly with windows (cygwin) slaves but not exclusively. My artifacts range from 1-2 mb to 800mb. Size appears not to be an issue.</p>

<p>At the time things hang (forever) both master and slave become quiet. Other jobs on other slaves progress without issue.</p>

<p>When I cancel the stuck build I see:</p>

<p>ERROR: Failed to archive artifacts: pool.tar.gz,*<b>/</b>.log,built.<b>,pool/</b><b>/</b>-Results.xml,<b>_pool.tar.gz,</b>.html,*.st<br/>
java.io.IOException: java.io.IOException: Failed to extract .....<br/>
<snip><br/>
Caused by: java.io.IOException<br/>
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:177)<br/>
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)<br/>
<snip></p>

<p>Will pfalcon's patch cure this?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>croftj:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=205779&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205779">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>And the beat goes on...</p>

<p>I am getting the same exception with Ubunto 12.04 remote and a Jenkins 1.509.1 master... sigh, I know it's an older master but still, yet another data point. This project worked and archived 25 days ago without any problem. Now it won't pass for nothing.</p>
</pre>
</details>

</ul>
</details>

<p>And the beat goes on...</p>

<p>I am getting the same exception with Ubunto 12.04 remote and a Jenkins 1.509.1 master... sigh, I know it's an older master but still, yet another data point. This project worked and archived 25 days ago without any problem. Now it won't pass for nothing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=205862&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205862">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is anyone experiencing this on Oracle JDK?</p>

<p>Quoting an earlier comment:</p>

<blockquote><p>We had a similar issue to this that was resolved by switching to Oracle's JDK from Open JDK.</p></blockquote>

<p><em>That</em> would be a real data point...</p>
</pre>
</details>

</ul>
</details>

<p>Is anyone experiencing this on Oracle JDK?</p>

<p>Quoting an earlier comment:</p>

<blockquote><p>We had a similar issue to this that was resolved by switching to Oracle's JDK from Open JDK.</p></blockquote>

<p><em>That</em> would be a real data point...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mbarczak:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=219507&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-219507">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Using Oracle vm doesn't seem to solve the problem. My master is Jenkins 1.580.2 on Ubuntu 14.04 (Oracle Java 1.8.0_25) , slaves are running Win Xp (Oracle Java 1.8.0_25). Changing the way that slaves are connecting (Windows Service -&gt; jnlp) didn't help neither. After rebooting affected slave, issue goes away for some time. What's interesting , when I run into situation with "archiving artifacts stuck" , I'm also not able to download anything from slave's workspace ( browsing the workspace works properly though)</p>
</pre>
</details>

</ul>
</details>

<p>Using Oracle vm doesn't seem to solve the problem. My master is Jenkins 1.580.2 on Ubuntu 14.04 (Oracle Java 1.8.0_25) , slaves are running Win Xp (Oracle Java 1.8.0_25). Changing the way that slaves are connecting (Windows Service -> jnlp) didn't help neither. After rebooting affected slave, issue goes away for some time. What's interesting , when I run into situation with "archiving artifacts stuck" , I'm also not able to download anything from slave's workspace ( browsing the workspace works properly though)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ssides.png" title="ssides's avatar" /> ssides:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=244600&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-244600">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>12/21/15: Update: using 1.642, I see the same as before. Job appears to have completed, and the last visible line (before killing the job) is 
"Archiving artifacts"
and there's a spinning icon below that.
This is not new. It has been an ongoing problem.
These are testing portion of jdk builds. The test runs are separate jobs. The one in question has run to completion and hangs while archiving(apparently).  Logs do not reveal anything helpful.
This happens on several different platforms, windows-x86, windowx-x64, solarix-x64, osx, linux-x86, sparc, linux-x64. It seems if I wait it out, jobs will eventually complete. It may take 2-3 or more hour (jobs usually take 15 minutes to 1 hr).</p>
</pre>
</details>

</ul>
</details>

<p>12/21/15: Update: using 1.642, I see the same as before. Job appears to have completed, and the last visible line (before killing the job) is <br/>
"Archiving artifacts"<br/>
and there's a spinning icon below that.<br/>
This is not new. It has been an ongoing problem.<br/>
These are testing portion of jdk builds. The test runs are separate jobs. The one in question has run to completion and hangs while archiving(apparently).  Logs do not reveal anything helpful.<br/>
This happens on several different platforms, windows-x86, windowx-x64, solarix-x64, osx, linux-x86, sparc, linux-x64. It seems if I wait it out, jobs will eventually complete. It may take 2-3 or more hour (jobs usually take 15 minutes to 1 hr).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=244602&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-244602">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is this issue limited to SSH slaves?</p>
</pre>
</details>

</ul>
</details>

<p>Is this issue limited to SSH slaves?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kavehv:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=251344&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-251344">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We seem to also be running into this issue using swarm slaves.</p>
</pre>
</details>

</ul>
</details>

<p>We seem to also be running into this issue using swarm slaves.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ottago:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=257567&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-257567">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We have just started experiencing the same problem, and its killing our build farm.  Its happening across platforms as mentioned before and does not seem specific to how the connection is launched.  Both ssh from the master or webstart from a slave have the same problem.</p>

<p>Some information that might help.  We don't always have artifacts, they are only present if tests fail and leave behind what they expected to be created which we can manually inspect later.  Could it be because it didn't find anything?</p>
</pre>
</details>

</ul>
</details>

<p>We have just started experiencing the same problem, and its killing our build farm.  Its happening across platforms as mentioned before and does not seem specific to how the connection is launched.  Both ssh from the master or webstart from a slave have the same problem.</p>

<p>Some information that might help.  We don't always have artifacts, they are only present if tests fail and leave behind what they expected to be created which we can manually inspect later.  Could it be because it didn't find anything?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jason_naylor:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=262973&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-262973">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm running into this now in version 1.651.3 on a windows slave using ssh
I'm not sure what is preventing the Archiving step from succeeding, but whatever it is leaves the build in a hung state indefinitely. This build agent has successfully built before and only started having this problem recently.</p>
</pre>
</details>

</ul>
</details>

<p>I'm running into this now in version 1.651.3 on a windows slave using ssh<br/>
I'm not sure what is preventing the Archiving step from succeeding, but whatever it is leaves the build in a hung state indefinitely. This build agent has successfully built before and only started having this problem recently.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jason_naylor:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=262976&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-262976">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The most disruptive part of this hang is that it disregards the build timeout. I have 'Abort the build if it is stuck' set with an absolute timeout of 45 minutes configured in the job and the build was stuck at the Archiving step for over 2 and a half hours.</p>
</pre>
</details>

</ul>
</details>

<p>The most disruptive part of this hang is that it disregards the build timeout. I have 'Abort the build if it is stuck' set with an absolute timeout of 45 minutes configured in the job and the build was stuck at the Archiving step for over 2 and a half hours.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>josephpattara:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=271573&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-271573">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We were hitting with this issue but atleast in our case I was able to get a work around as below and with my limited experience I doubt the actual problem is with the Jenkins code interaction with DNS.</p>

<p>In my case the Jenkins Master and the slave where I had the Artifact hanging was in different network domain. 
Example a simple ping command for me from jenkins master</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"></pre>
</div></div>
<p>ping windows_machine_with_problem
PING windows_machine_with_problem.DOMAIN1 (172.23.136.85) 56(84) bytes of data.
64 bytes from windows_machine_with_problem.DOMAIN1 (172.23.136.85): icmp_seq=1 ttl=127 time=0.287 ms
64 bytes from windows_machine_with_problem.DOMAIN1 (172.23.136.85): icmp_seq=2 ttl=127 time=0.335 ms
64 bytes from windows_machine_with_problem.DOMAIN1 (172.23.136.85): icmp_seq=3 ttl=127 time=0.377 ms</p>

<p>ping windows_machine_with_noproblem
PING windows_machine_with_noproblem.DOMAIN2 (172.28.8.87) 56(84) bytes of data.
64 bytes from 172.28.8.87: icmp_seq=1 ttl=128 time=0.379 ms
64 bytes from 172.28.8.87: icmp_seq=2 ttl=128 time=0.519 ms
64 bytes from 172.28.8.87: icmp_seq=3 ttl=128 time=0.400 ms</p>

<p>This indicated any traffic which goes from the master to windows_machine_with_problem passes through some networking entities.
To avoid any DNS from equation I just replaced the "hostname" in slave configuration directly with IP address and the artifact hanging disappeared.</p>

<p>So to summarize when I used ip-address instead of DNS name , atleast in my case artifact hanging issue was resolved.</p>

</pre>
</details>

</ul>
</details>

<p>We were hitting with this issue but atleast in our case I was able to get a work around as below and with my limited experience I doubt the actual problem is with the Jenkins code interaction with DNS.</p>

<p>In my case the Jenkins Master and the slave where I had the Artifact hanging was in different network domain. <br/>
Example a simple ping command for me from jenkins master</p>

<pre>
</pre>
<p>ping windows_machine_with_problem<br/>
PING windows_machine_with_problem.DOMAIN1 (172.23.136.85) 56(84) bytes of data.<br/>
64 bytes from windows_machine_with_problem.DOMAIN1 (172.23.136.85): icmp_seq=1 ttl=127 time=0.287 ms<br/>
64 bytes from windows_machine_with_problem.DOMAIN1 (172.23.136.85): icmp_seq=2 ttl=127 time=0.335 ms<br/>
64 bytes from windows_machine_with_problem.DOMAIN1 (172.23.136.85): icmp_seq=3 ttl=127 time=0.377 ms</p>

<p>ping windows_machine_with_noproblem<br/>
PING windows_machine_with_noproblem.DOMAIN2 (172.28.8.87) 56(84) bytes of data.<br/>
64 bytes from 172.28.8.87: icmp_seq=1 ttl=128 time=0.379 ms<br/>
64 bytes from 172.28.8.87: icmp_seq=2 ttl=128 time=0.519 ms<br/>
64 bytes from 172.28.8.87: icmp_seq=3 ttl=128 time=0.400 ms</p>

<p>This indicated any traffic which goes from the master to windows_machine_with_problem passes through some networking entities.<br/>
To avoid any DNS from equation I just replaced the "hostname" in slave configuration directly with IP address and the artifact hanging disappeared.</p>

<p>So to summarize when I used ip-address instead of DNS name , atleast in my case artifact hanging issue was resolved.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kavehv:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-7641?focusedId=272085&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-272085">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This still happens for us in jenkins 2.21 with a number of Ubuntu 14.04 slaves.</p>
</pre>
</details>

</ul>
</details>

<p>This still happens for us in jenkins 2.21 with a number of Ubuntu 14.04 slaves.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/13614">JENKINS-13614</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/15682">JENKINS-15682</a>]</i>
